src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -16,15 +16,25 @@ var ReactOwner = require('ReactOwner');
 
 import type { ReactInstance } from 'ReactInstanceType';
 import type { ReactElement } from 'ReactElementType';
+import type { TestRendererMockConfig } from 'ReactTestMount';
 
 var ReactRef = {};
 
-function attachRef(ref, component, owner) {
+function attachRef(ref, component, owner, mockConfig) {
```

Comment:
This is reconciler code, it also shouldn't have any knowledge about “mock config” (which is test renderer specific). We should *always* call `component.getPublicInstance()`. Passing an additional thing to it is fine, but that thing should also be renderer-agnostic (like `transaction`).

Instead of branching here, we should branch inside `getPublicInstance()` implementation in the test renderer. It also shouldn’t check if `mockConfig` exists because it should always exist—as we wanted to pass down a default value when user doesn’t specify any.